src/features/hotspot-onboarding/iot-ble/AddGatewayBle.tsx

```diff
@@ -0,0 +1,232 @@
+import BackScreen from '@components/BackScreen'
+import ButtonPressable from '@components/ButtonPressable'
+import CircleLoader from '@components/CircleLoader'
+import Text from '@components/Text'
+import {
+  init,
+  iotInfoKey,
+  keyToAssetKey,
+  rewardableEntityConfigKey,
+} from '@helium/helium-entity-manager-sdk'
+import {
+  AddGatewayV1,
+  useHotspotBle,
+  useOnboarding,
+} from '@helium/react-native-sdk'
+import {
+  bufferToTransaction,
+  heliumAddressToSolAddress,
+  sendAndConfirmWithRetry,
+} from '@helium/spl-utils'
+import { useNavigation } from '@react-navigation/native'
+import { LAMPORTS_PER_SOL, PublicKey, Transaction } from '@solana/web3.js'
+import { useAccountStorage } from '@storage/AccountStorageProvider'
+import { DAO_KEY, IOT_SUB_DAO_KEY } from '@utils/constants'
+import { getHotspotWithRewards } from '@utils/solanaUtils'
+import { Buffer } from 'buffer'
+import React from 'react'
+import { useAsyncCallback } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { Alert } from 'react-native'
+import { TabBarNavigationProp } from '../../../navigation/rootTypes'
+import { useSolana } from '../../../solana/SolanaProvider'
+import { CollectableNavigationProp } from '../../collectables/collectablesTypes'
+
+const AddGatewayBle = () => {
+  const { getOnboardingRecord, getOnboardTransactions, onboardingClient } =
+    useOnboarding()
+  const { createGatewayTxn, getOnboardingAddress } = useHotspotBle()
+  const { currentAccount } = useAccountStorage()
+  const { anchorProvider } = useSolana()
+  const { t } = useTranslation()
+  const tabNav = useNavigation<TabBarNavigationProp>()
+  const collectNav = useNavigation<CollectableNavigationProp>()
+
+  const {
+    execute: handleAddGateway,
+    loading,
+    error,
+  } = useAsyncCallback(async () => {
+    if (!anchorProvider) {
+      Alert.alert('Error', 'No anchor provider')
+      return
+    }
+    const accountAddress = currentAccount?.address
+    if (!accountAddress) {
+      Alert.alert(
+        'Error',
+        'You must first add a wallet address from the main menu',
+      )
+      return
+    }
+
+    const onboardAddress = await getOnboardingAddress()
+    const onboardRecord = await getOnboardingRecord(onboardAddress)
+
+    if (!onboardRecord) {
+      throw new Error(
+        `This hotspot does not exist in the onboarding server. Contact your manufacturer to have them approve hotspot with id ${onboardAddress}`,
+      )
+    }
+
+    if (!onboardRecord?.maker.address) {
+      throw new Error('Could not get maker address')
+    }
+    const makerSolAddr = heliumAddressToSolAddress(onboardRecord?.maker.address)
+    const makerSolBalance = (
+      await anchorProvider.connection.getAccountInfo(
+        new PublicKey(makerSolAddr),
+      )
+    )?.lamports
+    if (
+      !makerSolBalance ||
+      makerSolBalance / LAMPORTS_PER_SOL < 0.00089088 + 0.00001
+    ) {
+      throw new Error(
```

Comment:
Is there a scenario where the user would pay onboarding fees?

Reply:
Not yet. There might be if the defunct maker hip passes